htdocs/lib.js

```diff
@@ -0,0 +1,128 @@
+// Helper function to connect to the echo endpoint
+async function connect(pc) {
+    const {protocol} = window.location;
+    const ws = new WebSocket((protocol === 'http:' ? 'ws:' : 'wss:') + window.location.origin.substring(protocol.length) + '/ws');
+    ws.binaryType = 'arraybuffer';
+
+    await (new Promise((resolve) => {
+        ws.addEventListener('open', resolve);
+    }))
+    await pc.setLocalDescription();
+    // wait for ICE gathering to complete
+    await (new Promise((resolve) => {
+        if (pc.iceGatheringState === 'complete') {
+            resolve();
+        } else {
+            pc.addEventListener('icegatheringstatechange', function listener() {
+                if (pc.iceGatheringState === 'complete') {
+                    pc.removeEventListener('icegatheringstatechange', listener);
+                    resolve();
+                }
+            });
+        }
+    }));
+    const offer = pc.localDescription;
+    ws.send(JSON.stringify(offer));
+
+    const answer = await (new Promise((resolve) => {
+        ws.addEventListener('message', function listener(message) {
+            ws.removeEventListener('message', listener);
+            resolve(JSON.parse(message.data))
+        });
+    }));
+    await pc.setRemoteDescription(answer);
+    return ws;
+}
+
+// Determine if a packet is an RTCP packet.
+// See https://tools.ietf.org/html/rfc5761#section-4
+function isRTCP(packet) {
```

Comment:
This makes me think you maybe want the WebRTC endpoint to echo back RTCP too?

Reply:
I tried but somehow that ended up blocking. Then I found a bug in the JS client code here (missing `& 0x7f` on the payloadType here and things started working. Will update :-)